flake-module.nix

```
@@ -0,0 +1,83 @@
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mdDoc
+    mkOption
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.process-compose = mkOption {
+          description = mdDoc ''
+            process-compose-flake: creates process-compose executables
+            from your Nix attribute sets.
+          '';
+          type = types.submodule {
+            options = {
+              package = mkOption {
+                type = types.package;
+                default = pkgs.process-compose;
+              };
+              configs = mkOption {
+                type = types.attrsOf types.attrs;
```

`let yamlFormat = pkgs.formats.yaml { };`

and then

```suggestion
                type = types.attrsOf yamlFormat.type;
```

However, a submodule with options that duplicate the schema would be nicer. That way we get to document and improve the schema. This is good enough for a first release though.

---

We will duplicate the schema in a succeeding PR. I'm tracking this in #2